shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -717,6 +957,29 @@ private static PointerCoords parsePointerCoords(Object rawCoords, float density)
     return coords;
   }
 
+  // Creating a VirtualDisplay larger than the size of the device screen size
+  // could cause the device to restart: https://github.com/flutter/flutter/issues/28978
+  private void validateVirtualDisplayDimensions(int width, int height) {
+    DisplayMetrics metrics = context.getResources().getDisplayMetrics();
+    if (height > metrics.heightPixels || width > metrics.widthPixels) {
+      String message =
+          "Creating a virtual display of size: "
+              + "["
+              + width
+              + ", "
+              + height
+              + "] may result in problems"
+              + "(https://github.com/flutter/flutter/issues/2897)."
+              + "It is larger than the device screen size: "
+              + "["
+              + metrics.widthPixels
+              + ", "
+              + metrics.heightPixels
+              + "].";
+      Log.w(TAG, message);
+    }
+  }
```

Comment:
This seems like we should just clamp to the display size and print a warning that we did that...

Reply:
(I'm mainly concerned that the current failure mode is a potential crash on user devices that never shows up for the developer, as opposed to a potential resizing that happens on user devices)

Reply:
I did this in https://github.com/flutter/engine/commit/5a2c81b3d4613c4410c4237d331aeba865cdbd56